Sources/Kommunicate/Classes/Kommunicate.swift

```diff
@@ -593,7 +594,56 @@ open class Kommunicate: NSObject,Localizable, KMPreChatFormViewControllerDelegat
         }
     }
     
-    public func userSubmittedResponse(name: String, email: String, phoneNumber: String, password: String) {
+    /**
+    launches the PreChat Lead Collection with custom payload
+
+     - Parameters:
+     - appID: User's application ID.
+     - inputList: list of LeadCollectionField objects to create a form in pre Chat.
+     - viewController: ViewController from which the pre-chat form view will be launched.
+     */
+    open class func launchPreChatWithCustomPayload(appID: String,viewController: UIViewController, inputList: [LeadCollectionField],completion: @escaping([String:String]?,KommunicateError?) -> ()){
+        KMUserDefaultHandler.setApplicationKey(appID)
+        Kommunicate.presentingViewController = viewController
+        
+        Kommunicate.leadArray = inputList
+        
+        if !KMUserDefaultHandler.isLoggedIn() {
+            DispatchQueue.main.async {
+                if !Kommunicate.leadArray.isEmpty {
+                    let customPreChatVC = CustomPreChatFormViewController(configuration: Kommunicate.defaultConfiguration)
+                   
+                    customPreChatVC.submitButtonTapped = {  (response:[String:String]) in
+                        completion(response,nil)
+                        Kommunicate.presentingViewController.dismiss(animated: false, completion: nil)
+
+                    }
+                    customPreChatVC.closeButtonTapped = {
+                        Kommunicate().closeButtonTapped()
+                        completion(nil,.prechatFormNotFilled)
+                    }
+                    viewController.present(customPreChatVC, animated: false, completion: nil)
```

Comment:
Shouldn't we return the completion after presenting the VC instead of doing it after the close or submit button tap?

Reply:
we sending filled form response in the completion. so I guess we need to call completion on submit or close scenarios.

Reply:
No, completion shouldn't be used for that. Completion is being used to tell if PrechatVC has been shown. Accept another handler for user action like submit and close.

Reply:
okay